File: snippets/python_snippets/src/buy_btc.py
import logging
import breez_sdk




def buy(sdk_services):
    try:
        # ANCHOR: buy-btc
        req =  breez_sdk.BuyBitcoinRequest(breez_sdk.BuyBitcoinProvider.MOONPAY)
        buy_bitcoin_resp = sdk_services.buy_bitcoin(req=req)
        # ANCHOR_END: buy-btc
    except Exception as error:
        logging.error(error)
        raise

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/BuyBtc.kt
package com.example.kotlinmpplib

import breez_sdk.*

class BuyBtc {
    fun buy_btc(sdk: BlockingBreezServices) {
        // ANCHOR: buy-btc
        try {
            // Choose your provider
            val provider = BuyBitcoinProvider.MOONPAY
            // request the url to proceed with the Bitcoin acquisition
            val url = sdk.buyBitcoin(BuyBitcoinRequest(provider)).url
        } catch (e: Exception) {
            // Handle error
        }
        // ANCHOR_END: buy-btc
    }
}